Fix Canvas Colour Not Being Preserved by Script Generator

cailafinn opened the pull request:
**Description of work.**
Added the ability for the script generator to access and maintain the `facecolor` or canvas colour when generating a script from a plot. 

**To test:**
1. Plot something
2. Enter the figure options and change the canvas colour
3. Generate a script
4. Run the script
5. The shown plot should have preserved its background colour

Fixes #33225 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

Commit:
Preserve facecolor when outputting plot to script

RE #33225

Commit:
Add tests for facecolor functionality

RE #33225

Commit:
Amend existing tests to account for fix

RE #33225

Commit:
Add release notes

RE #33225

Harrietbrown reviewed (changes requested):
This does work from the script copied to clipborad, but project recovery still does not preserve the background colour. clould you check to see if it would be possible to fix that as part of this PR?

Commit:
Fix tests on Windows and macOS

Disconnect between inclusion of decimal point on default color means
that testing it is just testing matplotlib, which isn't what we care
about.

RE #33225

Commit:
Add facecolor options to project recovery

RE #33225

Commit:
Fix plotssaver and plotsloader tests

RE #33225

ConorMFinn commented on qt/python/mantidqt/mantidqt/project/test/test_plotssaver.py:
```diff
@@ -170,7 +170,7 @@ def test_get_dict_from_fig(self):
 
         self.loader_plot_dict[u'creationArguments'] = [[{u"specNum": 2, "function": "plot", u"normalize_by_bin_width":
-                                                                                            True}]]
+                                                                                            False}]]
```
I'm not clear on why this value changed. It seems that the sample workspace changed whether or not it was normalised to bin width when I added the lines to `plotssaver` and `plotsloader`. Some kind of undefined behaviour causing some unexpected consequence maybe?

Harrietbrown reviewed (approved):
The plot canvas background colour is preserved in the generated script as well as project recovery. Good Job
